tests/binary_counter/BUILD

```diff
@@ -14,16 +14,44 @@
 
 load("//platformio:platformio.bzl", "platformio_project")
 
-platformio_project(
-    name = "binary_counter",
+configs = [
+    (
+        "mega",
+        "megaatmega2560",
+        [
+            "-DRED_LED_PIN=44",
+            "-DORANGE_LED_PIN=46",
+            "-DGREEN_LED_PIN=48",
+            "-DBLUE_LED_PIN=50",
+            "-DYELLOW_LED_PIN=52",
+            "-DBUTTON_PIN=2",
+        ],
+    ),
+    (
+        "nano",
+        "nanoatmega328",
+        [
+            "-DRED_LED_PIN=3",
+            "-DORANGE_LED_PIN=4",
+            "-DGREEN_LED_PIN=5",
+            "-DBLUE_LED_PIN=6",
+            "-DYELLOW_LED_PIN=7",
+            "-DBUTTON_PIN=2",
+        ],
+    ),
+]
+
+[platformio_project(
+    name = "binary_counter_%s" % config[0],
     src = "binary_counter.cc",
-    board = "megaatmega2560",
+    board = config[1],
```

Comment:
Using indexation here is going to make it a bit challenging to map between the two. Did we consider making configs a dictionary with well chosen keys, so that these references become more self-documenting?

Reply:
I planned on making them structs, but bazel won't allow this. Dictionaries are a good idea, let me take a look

Reply:
This looks great, thank you.